crates/noirc_abi/src/lib.rs

```diff
@@ -120,6 +126,7 @@ impl AbiParameter {
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct Abi {
     pub parameters: Vec<AbiParameter>,
+    pub param_witnesses: BTreeMap<String, Vec<Witness>>,
```

Comment:
Can you add a comment on what this is? -- Ideally all public methods get comments, so its easier to do it as we add/modify things